packages/components/addon/components/hds/interactive/index.hbs

```diff
@@ -0,0 +1,41 @@
+{{#if @route}}
+  {{#if this.isRouteExternal}}
+    <LinkToExternal
+      @current-when={{@current-when}}
+      @models={{hds-link-to-models @model @models}}
+      @query={{hds-link-to-query @query}}
+      @replace={{@replace}}
+      @route={{@route}}
+      ...attributes
+    >
+      {{yield}}
+    </LinkToExternal>
+  {{else}}
+    <LinkTo
+      @current-when={{@current-when}}
+      @models={{hds-link-to-models @model @models}}
+      @query={{hds-link-to-query @query}}
+      @replace={{@replace}}
+      @route={{@route}}
+      ...attributes
+    >
+      {{yield}}
+    </LinkTo>
+  {{/if}}
+
+  {{! IMPORTANT: we can't support the direct use of the "href" HTML attribute via ...attributes, because we need to rely on the "@href" Ember argument to differentiate between different types of generated output }}
+{{else if @href}}
+  {{#if this.isHrefExternal}}
+    <a target="_blank" rel="noopener noreferrer" ...attributes href={{@href}}>
+      {{yield}}
+    </a>
+  {{else}}
+    <a ...attributes href={{@href}}>
+      {{yield}}
+    </a>
+  {{/if}}
+{{else}}
+  <button type="button" ...attributes>
```

Comment:
I like that we set this by default to `type="button"` to prevent accidental form submissions